app/controllers/organisations_controller.rb

```diff
@@ -30,17 +30,21 @@ def index
   def show
     if @organisation.nil?
       render template: 'pages/404', status: 404
+      return
```

Comment:
the changes in the show method accomplished two purposes: 1) reduce the nesting of "if/else" clauses, and 2) reduces the number of comparisons being made (many lines ended with "if current_user"; I substituted with one comparison and a block)